src/components/dialogs/AddPlayerDialogController.ts

```diff
@@ -20,8 +20,13 @@ export function addPlayer(): void {
 
     game.addPlayer(newPlayer);
 
-    // TODO: Only do this if the number of active players is less than the maximum number of active players
-    game.cycles[uiState.cycleIndex].addPlayerJoins(newPlayer);
+    const teamName: string = newPlayer.teamName;
+    const activePlayersCount: number = game.getActivePlayers(teamName, uiState.cycleIndex).size;
+    const maxActivePlayers: number = game.getMaxActivePlayers();
+
+    if (activePlayersCount < maxActivePlayers) {
+        game.cycles[uiState.cycleIndex].addPlayerJoins(newPlayer);
+    }
```

Comment:
If this is false, then we should let the user know why they weren't able to add a player. There might also be a case where we need to add a player first, then substitute them in for another (for example, a player joins in late, then gets subbed at the half).

Reply:
Perhaps adding a checkbox on the adding player dialogue that asks whether the added player should be a sub or not? If the activePlayersCount is equal to maxActivePlayers, then we force the checkbox as selected?

Reply:
`isStarter` should generally be false here. The one counterexample I can think of is if the player joins on cycle 0, but the join event should handle it correctly anyway, so I don't think we need to change anything there.

A checkbox makes sense to add. We also have to add a field to the `IPlayerJoinsEvent` and make it backward compatible. This means that if the field value doesn't exist, we should treat the player as being active. Therefore the variable should be `false` if they are active when they join; maybe something like `onBench` or `startsOnBench`.

You'll have to update `GameState.getActivePlayers` to look for this new flag.